docs/source/services/timelock_service/server_configuration.rst

```diff
@@ -0,0 +1,101 @@
+.. _timelock-server-configuration:
+
+Timelock Server Configuration
+=============================
+
+The Timelock Server configuration file is written in YAML and is located at ``var/conf/timelock.yml``.
+It has three main blocks: the ``clients`` block, ``cluster`` block and ``atomix`` block. We will discuss how each of
+these may be configured in turn, as well as additional configuration parameters.
+
+.. contents::
+   :local:
+
+Clients
+-------
+
+The ``clients`` block is a list of strings which corresponds to client namespaces that the server will respond to.
+At least one client must be configured for the Timelock Server to work. Querying an endpoint for a client that does not
+exist will result in a 404.
+
+   .. code:: yaml
+
+      clients:
+        - tom
+        - jerry
+
+A single Timelock Server or cluster of Timelock Servers can support multiple AtlasDB clients. When querying a
+Timelock Server, clients must supply a namespace for which they are requesting timestamps or locks in the form of a
+path variable. There are no guarantees of relationships between timestamps requested from different namespaces.
+
+   .. code:: bash
+
+      curl -XPOST localhost:8080/tom/timestamp/fresh-timestamp
+      curl -XPOST localhost:8080/jerry/timestamp/fresh-timestamp # no guarantees of any relationship between the values
+
+This is done for performance reasons: consider that if we maintained a global timestamp across all clients, then
+requests from each of these clients would need to all be synchronized.
+
+Cluster
+-------
+
+The ``cluster`` block is used to identify the servers which make up a Timelock Service cluster. An example is as
+follows:
+
+   .. code:: yaml
+
+      cluster:
+        localServer: http://palantir.com:8700
+        servers:
+          - http://palantir.com:8700
+          - http://palantir.com:8701
+
+.. list-table::
+   :widths: 5 40
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - localServer
+     - A string following the form ``protocol://hostname:port`` which matches the host on which this config exists.
+
+   * - servers
+     - A list of strings following the form ``protocol://hostname:port`` identifying the hosts in this Timelock
+       Service cluster. Note that this list must include the ``localServer``.
+
+Atomix
+------
+
+The Timelock Servers use the `Atomix <http://atomix.io>`__ library, and allow for some configuration as to
+Atomix-related communication and persistence. Note that unlike the ``clients`` and ``cluster`` blocks, this block is
+optional.
```

Comment:
I think it would be worth having example YAML here too. I really appreciate just an example I can look at, and just copy/paste into my own `timelock.yml`. Presumably it looks something like this?

```
atomix:
    storageLevel: DISK
    storageDirectory: var/data/atomix
...etc
```